stdlib/public/core/LegacyABI.swift

```diff
@@ -87,3 +87,11 @@ extension String.UTF16View {
   @inlinable @inline(__always)
   internal var _shortHeuristic: Int { return 32 }
 }
+
+extension _StringGuts {
+  @available(*, unavailable, renamed: "_StringGuts.init(immortal:isASCII:)")
+  @inlinable @inline(__always)
+  internal init(_ bufPtr: UnsafeBufferPointer<UInt8>, isASCII: Bool) {
+    self.init(_StringObject(immortal: bufPtr, isASCII: isASCII))
+  }
+}
```

Comment:
Is it OK to make this _unavailable_?

It was used by [`String.init(_builtinStringLiteral:utf8CodeUnitCount:isASCII:)`](https://github.com/apple/swift/blob/c6cd4b07b2654cadc28cd16468e98bd2aac9e58b/stdlib/public/core/String.swift#L638-L655)

Reply:
CC @lorentey , does `unavailable` impact the ABI and compatibility story?

Reply:
It doesn't -- this will still get compiled into the stdlib binary. I recommend making this `@usableFromInline` instead of leaving it inlinable though. New code won't be able to call this anyway, so there is little point in continuing to expose its body.

Reply:
@lorentey Thanks. The ABI stability test currently needs:

```diff
+Constructor _StringGuts.init(_:isASCII:) has been removed
```

Do you think this should be reworded? (Since it hasn't been *removed* from the ABI.)

Reply:
Hm. This feels like a false positive (from an ABI viewpoint) -- unless I'm wrong about the entry point still getting compiled into the binary. (To double check, see if the corresponding mangled symbol is still exported by libswiftCore.dylib with these changes. It should be still there!)

(Cc @xwu)

Reply:
@lorentey I've updated *everything* in LegacyABI.swift to be `@available(*, unavailable)` and `@usableFromInline`.

After a clean build:

```sh
cd build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64/lib/swift/macosx
xcrun dyldinfo -export libswiftCore.dylib
```

The export information (for x86_64 and arm64) contains:

* `_$ss11_StringGutsV_7isASCIIABSRys5UInt8VG_SbtcfC`
* `_$sSS25_nativeCopyUTF16CodeUnits4into5rangeySrys6UInt16VG_SnySS5IndexVGtF`
* `_$ss11_branchHint_8expectedS2b_SbtF`
* `_$sSS9UTF16ViewV15_shortHeuristicSivg`
* `_$sSS9UTF16ViewV15_shortHeuristicSivpMV`

(These are the mangled names of the APIs I made unavailable.)